### PR TITLE
Add AlignValueAttribute

### PR DESCRIPTION
**Pull request by isuruf**
See https://github.com/inducer/loopy/pull/707 and https://reviews.llvm.org/D4635 for context

### CODE REVIEW

File: cgen/__init__.py [hunk 1]
```diff
@@ -295,6 +295,18 @@ def get_decl_pair(self):
     mapper_method = "map_aligned"
 
 
+class AlignValueAttribute(NestedDeclarator):
```
Comment (inducer) [inline cgen/__init__.py:298]: Add a references/context:

- https://www.intel.com/content/www/us/en/develop/documentation/oneapi-dpcpp-cpp-compiler-dev-guide-and-reference/top/compiler-reference/attributes/align-value.html
- https://reviews.llvm.org/D4635

maybe also explaining the difference with the "normal" `aligned` attribute. (which I'm still not sure I understand) Reading the [GCC docs](https://gcc.gnu.org/onlinedocs/gcc/C-Extensions.html#C-Extensions) suggests that the attribute applies to the whole variable no matter where in the declarator it occurs, though I'm not totally sure about that.

Comment (isuruf) [reply cgen/__init__.py:301]: Done in 1ddbb6752fccb1bb1335f9c08be44078fb223c1d

Comment (inducer) [reply cgen/__init__.py:301]: Nice, thanks!

Comment (inducer) [reply cgen/__init__.py:301]: Oh, almost forgot: please hook this into the docs.

**Comment by inducer** [comment]
Thanks!

**Pull request merged**